crates/control/src/behavior/node.rs

```diff
@@ -275,6 +279,51 @@ impl Behavior {
             .dribble_path_obstacles_output
             .fill_if_subscribed(|| dribble_path_obstacles.clone().unwrap_or_default());
 
+        if let (Some(kick_decisions), Some(instant_kick_decisions)) = (
+            world_state.kick_decisions.as_ref(),
+            world_state.instant_kick_decisions.as_ref(),
+        ) {
+            // let kick_decisions = world_state.kick_decisions.as_ref().unwrap();
+            // let instant_kick_decisions = world_state.instant_kick_decisions.as_ref().unwrap();
+
+            let available_kick_bigger = kick_decisions
+                .iter()
+                .chain(instant_kick_decisions.iter())
+                .find(|decision| {
+                    dribble::is_kick_pose_reached(
+                        decision.kick_pose,
+                        context.in_walk_kicks[decision.variant].reached_thresholds,
+                        world_state.robot.ground_to_upcoming_support,
+                    )
+                });
+            let available_kick_smaller = kick_decisions
+                .iter()
+                .chain(instant_kick_decisions.iter())
+                .find(|decision| {
+                    dribble::is_kick_pose_reached(
+                        decision.kick_pose,
+                        context.in_walk_kicks[decision.variant].precision_kick_reached_thresholds,
+                        world_state.robot.ground_to_upcoming_support,
+                    )
+                });
+            if available_kick_bigger.is_some()
+                && available_kick_smaller.is_none()
```

Comment:
Please use `if let`

Reply:
i think `match` is better in this situation because of the `else if`
done